classes/lang/DataLang.php

```diff
@@ -46,22 +48,13 @@ public function __construct($locale)
     {
         $this->locale = $locale;
 
-        $legacyTranslator = Context::getContext()->getTranslator();
-        $legacyLocale = $legacyTranslator->getLocale();
+        $this->translator = Context::getContext()->getTranslator();
+        $translatorLocale = $this->translator->getLocale();
 
-        if ($legacyLocale === $this->locale) {
-            $this->translator = $legacyTranslator;
-        } else {
-            $this->translator = new Translator(
-                $this->locale,
-                null,
-                _PS_CACHE_DIR_ . '/translations/' . $this->locale,
-                false
-            );
+        if ($translatorLocale !== $this->locale) {
```

Comment:
Waow! So fixing "only" this allows to manage the new language correctly?!

Reply:
Yes... Took time to find it though 😅
I just nedd to follow @eternoendless's recommandation by adding a new `TranslatorLanguageLoader` and we should be good with that PR